e2e/testsuite/semver.go

```diff
@@ -0,0 +1,75 @@
+package testsuite
+
+import (
+	"strconv"
+	"strings"
+)
+
+// Version follows semantic versioning while disregarding the patch version.
+type Version struct {
+	Major uint64
+	Minor uint64
+}
+
+// NewVersion constructs a new version given the major and minor version.
+func NewVersion(major, minor uint64) Version {
+	return Version{
+		Major: major,
+		Minor: minor,
+	}
+}
+
+func ParseChainVersion(version string) (Version, bool) {
```

Comment:
Is this function is only used in `IsSupported`, can it be unexported? If it cannot, can we add a godoc then?